### PR TITLE
IGNITE-16677 Implement broadcast method of IgniteCompute interface

### PR DESCRIPTION
**Pull request by rpuch**
https://issues.apache.org/jira/browse/IGNITE-16677

### CODE REVIEW

File: modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java [hunk 1]
```diff
@@ -68,14 +86,25 @@ private boolean isLocal(ClusterNode targetNode) {
...
-        int nodesToSkip = random.nextInt(nodes.size());
+    /** {@inheritDoc} */
+    @Override
+    public <R> Map<ClusterNode, CompletableFuture<R>> broadcast(
+            Set<ClusterNode> nodes,
+            Class<? extends ComputeJob<R>> jobClass,
+            Object... args
+    ) {
+        Map<ClusterNode, CompletableFuture<R>> resultMap = nodes.stream()
+                .collect(toMap(node -> node, node -> executeOnOneNode(node, jobClass, args)));
```
Comment (SammyVimes) [inline modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java:97]: You can use toUnmodifiableMap here

File: modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java [hunk 2]
```diff
@@ -68,14 +86,25 @@ private boolean isLocal(ClusterNode targetNode) {
...
-        }
+        return Map.copyOf(resultMap);
+    }
 
-        return iterator.next();
+    /** {@inheritDoc} */
+    @Override
+    public <R> Map<ClusterNode, CompletableFuture<R>> broadcast(Set<ClusterNode> nodes, String jobClassName, Object... args) {
+        Map<ClusterNode, CompletableFuture<R>> resultMap = nodes.stream()
+                .collect(toMap(node -> node, node -> executeOnOneNode(node, jobClassName, args)));
```
Comment (SammyVimes) [inline modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java:106]: Same as above

**Review by SammyVimes — CHANGES_REQUESTED** [review]

File: modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java (hunk 1 above)
Comment (rpuch) [reply modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java]: Fixed

File: modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java (hunk 2 above)
Comment (rpuch) [reply modules/compute/src/main/java/org/apache/ignite/internal/compute/IgniteComputeImpl.java]: Fixed

**Review by SammyVimes — APPROVED** [review]
LGTM!

**Comment by SammyVimes** [comment]
Merged to the main branch

**Pull request closed without merging**